lib/active_merchant/billing/gateways/cyber_source_rest.rb

```diff
@@ -258,56 +258,43 @@ def add_merchant_description(post, options)
       end
 
       def add_stored_credentials(post, payment, options)
-        return unless stored_credential = options[:stored_credential]
+        return unless options[:stored_credential]
 
-        options = stored_credential_options(stored_credential, options.fetch(:reason_code, ''))
-        post[:processingInformation][:commerceIndicator] = options.fetch(:transaction_type, 'internet')
-        stored_credential[:initial_transaction] ? initial_transaction(post, options) : subsequent_transaction(post, options)
+        post[:processingInformation][:commerceIndicator] = commerce_indicator(options.dig(:stored_credential, :reason_type))
+        add_authorization_options(post, options)
       end
 
-      def stored_credential_options(options, reason_code)
-        transaction_type = options[:reason_type]
-        transaction_type = 'install' if transaction_type == 'installment'
-        initiator = options[:initiator] if  options[:initiator]
-        initiator = 'customer' if initiator == 'cardholder'
-        stored_on_file = options[:reason_type] == 'recurring'
-        options.merge({
-          transaction_type: transaction_type,
-          initiator: initiator,
-          reason_code: reason_code,
-          stored_on_file: stored_on_file
-        })
+      def commerce_indicator(reason_type)
+        case reason_type
+        when 'recurring'
+          'recurring'
+        when 'installment'
+          'install'
+        else
+          'internet'
+        end
       end
 
-      def add_processing_information(initiator, merchant_initiated_transaction_hash = {})
-        {
+      def add_authorization_options(post, options)
+        initiator = options.dig(:stored_credential, :initiator) == 'cardholder' ? 'customer' : 'merchant'
+        authorization_options = {
           authorizationOptions: {
             initiator: {
               type: initiator,
-              merchantInitiatedTransaction: merchant_initiated_transaction_hash,
-              storedCredentialUsed: true
+              storedCredentialUsed: true,
```

Comment:
Is storedCredentialUsed supposed to be only for subsequent transactions?

Reply:
Their docs indicate this:
> true means the merchant-initiated transaction came from a card that was already stored on file.
false means the merchant-initiated transaction came from a card that was not stored on file.

Re-reading it I think I'm going to add a check that only includes this if it's an MIT!

Reply:
Can you please share the link to the docs?

Reply:
https://developer.cybersource.com/api-reference-assets/index.html#payments_payments_process-a-payment